worldedit-core/src/main/java/com/sk89q/worldedit/util/asset/holder/ImageHeightmap.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * WorldEdit, a Minecraft world manipulation toolkit
+ * Copyright (C) sk89q <http://www.sk89q.com>
+ * Copyright (C) WorldEdit team and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldedit.util.asset.holder;
+
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+
+public class ImageHeightmap {
+
+    private final BufferedImage image;
+
+    private BufferedImage resizedImage;
+    private int lastSize = -1;
+
+    public ImageHeightmap(BufferedImage image) {
+        this.image = image;
+    }
+
+    public double getHeightAt(int x, int y, int size, double intensity) {
+        if (size != lastSize || resizedImage == null) {
```

Comment:
I'm not sure this resize is worth it. Can't you just compute this when accessing the RGB data? e.g.:
```java
double ratioX = image.getWidth() / (double) size;
double ratioY = image.getHeight() / (double) size;

int rgb = image.getRGB((int) (ratioX * x), (int) (ratioY * y));
```

Reply:
That was from KennyTV's code, my assumption there was that resizing it allows for better gradients compared to just directly accessing. Realistically that's probably not a concern though as the image is likely much larger than the brush itself